### PR TITLE
F t29878 disconnect addon data storage

### PR DESCRIPTION
**Pull request by sierrazamudiodemos**
**Ticket:** https://yaits.demos-deutschland.de/T29878

Description:
- We want to remove the dependency between DemosCore and addons
- DemosCoreEntities should not be used directly in demospipes neither maillane. Instead, interfaces provided by the middlelayer addons should be used.
- This PR is about linking the DemosCoreEntities to its corresponding Interfaces in the middlelayer Addon. 

How to review/test
- This PR is related to demospipes and maillane
- The 3 PRs should be merged to be able to test

Delete the checkbox if it doesn't apply/isn't necessary.
- [ x] Move the tickets on the board accordingly

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## UNRELEASED
 ## v0.21 (2024-01-23)
-
+- Link references of DemosCoreEntities to Interfaces so that addons can use the interfaces
```
Comment (breierDemos) [inline CHANGELOG.md:5]: This line has to be under ## UNRELEASED not under ## v0.21 (2024-01-23)

File: src/DependencyInjection/DemosPlanAddonExtension.php [hunk 2]
```diff
@@ -8,8 +8,9 @@
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Extension\Extension;
 use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\DependencyInjection\Extension\PrependExtensionInterface;
 
-class DemosPlanAddonExtension extends Extension
+class DemosPlanAddonExtension extends Extension implements PrependExtensionInterface
```
Comment (breierDemos) [inline src/DependencyInjection/DemosPlanAddonExtension.php:13]: Maybe we must extend the AbstractAddonExtension?

**Review by breierDemos — CHANGES_REQUESTED** [review]

File: src/DependencyInjection/DemosPlanAddonExtension.php [hunk 3]
```diff
@@ -20,4 +21,21 @@ public function load(array $configs, ContainerBuilder $container)
         );
         $loader->load('services.yml');
     }
+
+    public function prepend(ContainerBuilder $container)
+    {
+        $container->loadFromExtension('doctrine', array(
+            'orm' => array(
+                'resolve_target_entities' => array(
+                    'DemosEurope\DemosplanAddon\Contracts\Entities\ProcedureInterface' => 'demosplan\DemosPlanCoreBundle\Entity\Procedure\Procedure',
```
Comment (bakridemos) [inline src/DependencyInjection/DemosPlanAddonExtension.php]: we can not use hier something like 'demosplan\...', actually we want to remove them from the addons that's why i recommended to add these definitions in the doctrine yaml in core

**Review by bakridemos — CHANGES_REQUESTED** [review]

File: src/DependencyInjection/DemosPlanAddonExtension.php (hunk 3 above)
Comment (sierrazamudiodemos) [reply src/DependencyInjection/DemosPlanAddonExtension.php]: hey @bakridemos this is the middle layer between each addon (demospipes, maillane etc) and demos core. As a middle layer, it knows about both projects, that is why I decided to do it here. I also checked it with Sami. What do you think? We could also review with sami this approach :)

Comment (MoritzMandler) [reply src/DependencyInjection/DemosPlanAddonExtension.php]: I thought the core and the addon both know about the middle layer - but the middle layer itself should not reference any
of them.

Comment (sierrazamudiodemos) [reply src/DependencyInjection/DemosPlanAddonExtension.php]: 1. the middlelayer has already some refereces to Core, for example in services file: https://github.com/demos-europe/demosplan-addon/blob/f_T29878_disconnect-addon-data-storage/config/services.yml

Yeah I see your point.

**Review by sierrazamudiodemos — COMMENTED** [review]
@breierDemos here 49c8b73529f5fb3a6423894edbf1818ef8560dce the commit for the UuidV4Generator. Addons demospipes, maillane and oracle-sgements will use this class instead of demoscore

File: src/Contracts/Doctrine/UuidV4Generator.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+use Doctrine\ORM\Id\AbstractIdGenerator;
+use Ramsey\Uuid\Uuid;
+
+class UuidV4Generator extends AbstractIdGenerator
+{
+    public function generateId(EntityManagerInterface $em, $entity): string
+    {
+        return Uuid::uuid4()->toString();
+    }
+}
```
Comment (breierDemos) [inline src/Contracts/Doctrine/UuidV4Generator.php:17]:
```suggestion
}

```

**Review by breierDemos — CHANGES_REQUESTED** [review]

**Review by breierDemos — APPROVED** [review]

File: src/DependencyInjection/DemosPlanAddonExtension.php (hunk 3 above)
Comment (bakridemos) [reply src/DependencyInjection/DemosPlanAddonExtension.php]: good catch, actualy i think we don't need these two definitions here, and i think we can remove them from there. It's done in https://github.com/demos-europe/demosplan-addon/pull/88

**Review by mussbach — APPROVED** [review]
lgtm

**Review by mussbach — CHANGES_REQUESTED** [review]
the location needs to be adjusted. This is no contract

**Review by mussbach — APPROVED** [review]
lgtm, thx. Code review only

**Review by breierDemos — APPROVED** [review]

**Pull request merged**